yarn-project/simulator/src/public/public_processor.ts

```diff
@@ -217,7 +218,9 @@ export class PublicProcessor {
       : await this.worldStateDB.storageRead(feeJuiceAddress, balanceSlot);
 
     if (balance.lt(txFee)) {
-      throw new Error(`Not enough balance for fee payer to pay for transaction (got ${balance} needs ${txFee})`);
```

Comment:
We should add an issue to move this check to when the transaction arrives?? In the p2p pool there is validation logic that should probably check this here, such that txs that are unable to cover the fee get dropped and not gossiped? 

We can make an issue to add this rather than adding in this pr